src/log_format_impls.cc

```diff
@@ -146,7 +146,7 @@ class generic_log_format : public log_format {
                 &level)) != NULL) {
             const char *level_str = &sbr.get_data()[level.c_begin];
             logline::level_t level_val = logline::string2level(
-                    level_str, level.length());
+                    level_str, strlen(level_str));
```

Comment:
So I started up lnav today and all the examples that I mentioned above started working and there haven't really been any check-ins since. Just taking a cursory look at the code here's the hypothesis, the `log_scanf` method will try and match the string against the known formats. If one of them matches, the values for `level.c_begin` and `level.c_end` get set otherwise they end up with garbage values and `level.length()` may or may not come out to be large enough to contain the generic level regex match that `string2level` tries to do with `LEVEL_RE`.